packages/map-template/src/components/LegendDialog/LegendDialog.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useRef, useState } from "react";
+import './LegendDialog.scss';
+import { useRecoilState, useRecoilValue } from "recoil";
+import primaryColorState from "../../atoms/primaryColorState";
+import isLegendDialogVisibleState from "../../atoms/isLegendDialogVisibleState";
+import { useTranslation } from "react-i18next";
+import { useIsKioskContext } from "../../hooks/useIsKioskContext";
+import legendSortedFieldsSelector from "../../selectors/legendSortedFieldsSelector";
+
+/**
+ * Handle the Legend dialog.
+ *
+ */
+function LegendDialog() {
+    const { t } = useTranslation();
+    const primaryColorProp = useRecoilValue(primaryColorState);
+
+    const legendSections = useRecoilValue(legendSortedFieldsSelector);
+
+    const isKioskContext = useIsKioskContext();
+
+    const legendModalRef = useRef();
+
+    const legendSectionRef = useRef();
+
+    const scrollButtonsRef = useRef();
+
+    const [showLegendDialog, setShowLegendDialog] = useRecoilState(isLegendDialogVisibleState);
+
+    const [isLegendScrollable, setIsLegendScrollable] = useState(false);
+
+    const legendSectionsRef = useRef();
+
+    /*
+     * Get the height of the legend sections. 
+     * Set the legend size atom and determine 
+     * if the scroll buttons should be shown.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext) {
+            legendSectionRef.current.clientHeight > 700 ? setIsLegendScrollable(true) : setIsLegendScrollable(false);
+        }
+    }, [showLegendDialog, isKioskContext]);
+
+    /*
+     * Setup scroll buttons to scroll in legend sections when in kiosk mode.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext && isLegendScrollable) {
+            const legendContent = document.querySelector('.legend__sections');
+            scrollButtonsRef.current.scrollContainerElementRef = legendContent;
+        }
+    }, [showLegendDialog, legendSections, isLegendScrollable]);
+
+    return (<>
+        <div className="legend__background"></div>
+        <div className="legend" ref={legendModalRef}>
+            {legendSections.length > 0 &&
+                <div className={`legend__sections ${isLegendScrollable ? 'legend__sections--scrollable' : ''}`}
+                    ref={legendSectionsRef}
+                    style={{ maxHeight: isLegendScrollable ? '700px' : 'auto', overflowY: isLegendScrollable ? 'scroll' : 'auto' }}>
```

Comment:
The value of 700 is used in more than one case. Consider using a variable.

Reply:
Good idea, it's used in two cases, once just `700` and once `700px`. Does it make sense to use a variable in this case?

Reply:
Sure. 

We can always use template literals to get the px string: `${myAwesomeVariable}px`.

Reply:
🙌